wasmtime/_bindings.py

```diff
@@ -2937,9 +2937,9 @@ def wasmtime_linker_instantiate_pre(linker: Any, module: Any, instance_pre: Any)
 
 _wasmtime_memorytype_new = dll.wasmtime_memorytype_new
 _wasmtime_memorytype_new.restype = POINTER(wasm_memorytype_t)
-_wasmtime_memorytype_new.argtypes = [c_uint64, c_bool, c_uint64, c_bool]
-def wasmtime_memorytype_new(min: Any, max_present: Any, max: Any, is_64: Any) -> ctypes._Pointer:
-    return _wasmtime_memorytype_new(min, max_present, max, is_64)  # type: ignore
+_wasmtime_memorytype_new.argtypes = [c_uint64, c_bool, c_uint64, c_bool, c_bool]
+def wasmtime_memorytype_new(min: Any, max_present: Any, max: Any, is_64: Any, shared: Any) -> ctypes._Pointer:
+    return _wasmtime_memorytype_new(min, max_present, max, is_64, shared)  # type: ignore
```

Comment:
FWIW this `bindings.py` is auto-generated meaning that you can't modify it directly. This reflects the headers themselves and I think it's a bug in the C API that you can't create a new shared memory type. I think that a function such as `wasmtime_sharedmemorytype_new` will need to be added to the C API (or a `shared: bool` flag to the existing constructor here)

Reply:
Yes, I had to add the `bool shared` there and some more glue code that was missing. Let me PR there and restore this file.